### DISCUSSION TITLE
Help with #44

### DISCUSSION THREAD

**Discussion by duarch** [cssbattle/forum#12] [General]
Please guys I'm struggling with this solution. I Know it's not a big deal but here is the code:

```
<p><p><p><p><p><dt a><dt>
<style>
  dt{
    border-radius:50%;
    border:50vh solid#0000;
    --:-300 210 0-210
  }
  [a]{
    --:-260 190
  }
  p{
    padding:10;
    background:#F3AC3C;
    --:20
  }
  *{
    background:#1A4341;
    margin:var(--,30)
  }
```
Using the preview it's almost impossible to notice some difference.

![image](https://user-images.githubusercontent.com/42032253/110219924-e82c0c00-7ea0-11eb-969c-0bb0dc58c595.png)

**Comment by chinchang** [comment]
@duarch There seems to be a few pixels that are not the right color in the bottom-most row of the image. You might wanna check them with a color picker

**Reply by duarch** [reply]
@chinchang Thank you but I didn't figure it out how to solve. I'm pausing this #44 for a moment

**Reply by gunnarbittersmann** [reply]
BTW, you don’t need the `a` attribute on the `dt` element. There’s another way to select this thing that’s gonna save you one character.

**Reply by duarch** [reply]
> BTW, you don’t need the `a` attribute on the `dt` element. There’s another way to select this thing that’s gonna save you one character.

Please teach me. 

By the way I saw a lot of uses of `*, * *, *>*, *+p` But actually I don't understand the correct use of them. Sometimes I tried to use it but <dt> behaves differently from <p>, so I leave it.

**Reply by gunnarbittersmann** [reply]
`p~dt` selects all `dt` elements that are following siblings to a `p` element. ([general sibling combinator](https://developer.mozilla.org/en-US/docs/Web/CSS/General_sibling_combinator))
`p+dt` selects a `dt` element that is the next sibling to a `p` element. ([adjacent sibling combinator](https://developer.mozilla.org/en-US/docs/Web/CSS/Adjacent_sibling_combinator))

`p a` selects all `a` elements that are descendants (i.e. inside) of a `p` element. ([descendant combinator](https://developer.mozilla.org/en-US/docs/Web/CSS/Descendant_combinator))
`p>a` selects all `a` elements that are children (i.e. immediately inside) of a `p` element. ([child combinator](https://developer.mozilla.org/en-US/docs/Web/CSS/Child_combinator))

**Reply by rafehimself** [reply]
@gunnarbittersmann 
is there a shortcut to select only first element of same type and omit others ? , like for example `<p><p><p><p>`
I need that to be applied to only first `<p>`, :first-of-type is too long 🤣

**Reply by gunnarbittersmann** [reply]
Like for example on [target #81](https://cssbattle.dev/play/81)? 😉

You could declare the property for all `p` and overwrite for all but the first one using the general or adjacent sibling combinator: `p{foo:bar}p~p{foo:baz}` or `p{foo:bar}p+p{foo:baz}`. That’s still too long?

Yes, it surely is. Set an attribute on the desired element: `<p a><p><p><p>` and style using the attribute selector. Never use classes when code-golfing.

**Comment by gunnarbittersmann** [comment]
Your solution should work. I  bet it would have with the old backend.
Me thinks it’s the anti-aliasing of the green circles over the yellow stripes. When using other properties instead of `border` and `border-radius`, I get a 100% match.

Fun thing is, on [target #29](https://cssbattle.dev/play/29) I got 100% easily with `border` and `border-radius`, but it was hard to achieve 100% with what works fine here. Maybe because foreground and background colors are flipped?

**Reply by himanshubhatt1** [reply]
but it's still not working

**Reply by SelenIT** [reply]
It seems that different ways to create circular shape use different algorithms for anti-aliasing edges, and rounding errors of some of them result in too much mismatching pixels, especially in places where the curve touches another line making a sharp egde. In this case, `border-radius` seems to create the "wrong type" of the circle, however, `radial-gradient` works:
```
<p><p><p><p><p><dt><dt>
<style>
  dt{
    padding:150;
    background:radial-gradient(#1A4341 159q,#0000 0);
    --:-260 210 0-210
  }
  dt+dt{
    --:-300 190
  }
  p{
    height:20;
    background:#F3AC3C;
    --:20
  }
  *{
    background:#1A4341;
    margin:var(--,30)
  }
```

**Reply by himanshubhatt1** [reply]
thanks bro, its working

**Reply by APerricone** [reply]
it is a very serious bug, why they don't update the backend?
